### ISSUE TITLE
Non optional reservation status on update operation

### ISSUE THREAD

**Issue by mcelicalderon** [twilio/twilio-php#330]
**Version:** 5.0.0-RC7
**API Subdomain (api/taskrouter/ip_messaging):** Taskrouter

[This method](https://github.com/twilio/twilio-php/blob/next-gen/Twilio/Rest/Taskrouter/V1/Workspace/Task/ReservationContext.php#L69) won't allow the user to update only the needed parameters without passing first the reservation status. 

As it is not optional, currently we have to fetch the instance to get the property so it remains unchanged after the update operation:

```php
$reservation->update(
    $reservation->reservationStatus,
    array(
        'instruction' => 'call',
        'callFrom' => '+15558675309',
        'callUrl' => 'http://example.com/agent_answer',
        'callStatusCallbackUrl' => 'http://example.com/agent_answer_status_callback',
        'callAccept' => 'true'
    )
);
```

**Comment by ihumanable** [comment]
According to https://www.twilio.com/docs/api/taskrouter/reservations#action-update ReservationStatus is a required parameter for update.

**Comment by ihumanable** [comment]
Misread our documentation, this does seem to be optional for some of the operations on POST to instance, will update definitions to reflect this.

**Comment by jingming** [comment]
Released with `5.0.0-RC9`.
